### PR TITLE
Corrected testcase :ClusterBackupShareToggle

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Updated the testcase to get the backup created by this testcase to validate 
Until now, since its cluster level share ,all the backup were shared from admin to user.
Incase there are some stale backup from previous testcase, this stale backup was also being shared and was sometimes used in BackupSuccessCheckWithValidation leading to failure as context to stale backup was not found.
**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PB-5888

**Special notes for your reviewer**:

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4924/console

https://aetos.pwx.purestorage.com/resultSet/testSetID/589514/testCaseID/3593989

### CODE REVIEW

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by ak-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

File: tests/backup/backup_share_test.go [hunk 1]
```diff
@@ -2358,6 +2359,9 @@ var _ = Describe("{ClusterBackupShareToggle}", func() {
 			_, err = CreateScheduleBackupWithValidation(ctx, scheduleName, backupClusterName, backupLocationName, backupLocationUID, scheduledAppContexts, labelSelectors, BackupOrgID, "", "", "", "", periodicSchedulePolicyName, periodicSchedulePolicyUid)
 			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of schedule backup with schedule name [%s]", scheduleName))
 			scheduleNames = append(scheduleNames, scheduleName)
+			log.InfoD("Getting the first schedule backup from admin")
```
Comment (kphalgun-px) [inline tests/backup/backup_share_test.go:2362]: I think CreateScheduleBackupWithValidation returns firstScheduleBackupName.

File: tests/backup/backup_share_test.go [hunk 2]
```diff
@@ -2393,14 +2399,16 @@ var _ = Describe("{ClusterBackupShareToggle}", func() {
 				}
 				log.InfoD("Waiting for 15 minutes for the next schedule backup to be triggered")
 				time.Sleep(15 * time.Minute)
+				latestScheduleBkpName, err := GetLatestScheduleBackupName(ctx, scheduleName, BackupOrgID)
```
Comment (kphalgun-px) [inline tests/backup/backup_share_test.go:2402]: The GetLatestScheduleBackupName does not necessarily return to the second backup name. You might want to use GetOrdinalScheduleBackupName instead.

```go
secondSchBackupName, err := GetOrdinalScheduleBackupName(ctx, scheduleName, 2, BackupOrgID)
dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching the name of the second schedule backup [%s]", secondSchBackupName))
```

File: tests/backup/backup_share_test.go (hunk 1 above)
Comment (snigdha-px) [reply tests/backup/backup_share_test.go:2362]: will take this in next PR since this is merged

File: tests/backup/backup_share_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_share_test.go:2402]: Yes yes, here I need to get the latest schedule backup only, not the second one

**Pull request merged**